+Prefs/Double.m

```diff
@@ -35,15 +36,18 @@ function set_ui_value(obj,val)
             end
         end
         function validate(obj,val)
-            validateattributes(val,{'numeric'},{'scalar'})
+            if ~isnumeric(val) || ~isscalar(val)
+                error('Double must be numeric and scalar.')
+            end
             if ~obj.allow_nan
                 assert(~isnan(val),'Attempted to set NaN. allow_nan is set to false.')
             elseif isnan(val)
                 return
             end
-            assert(val <= obj.max, 'Cannot set value greater than max.')
-            assert(val >= obj.min, 'Cannot set value less than min.')
+            if val <= obj.max || val >= obj.min
```

Comment:
why not just turn this into an error statement if you don't want to use the validate method?

Reply:
assert evaluates the string whether or not the assertion is true or false. This was limiting performance. Now the strings are generated only if they need to be.